tools/ld-dropout-correct/dropoutcorrect.cpp

```diff
@@ -294,83 +327,124 @@ DropOutCorrect::Replacement DropOutCorrect::findReplacementLine(const QVector<Dr
-                                 firstActiveFieldLine, lastActiveFieldLine,
-                                 candidates);
-
-    // Only check the other field for visible line replacements
-    if (!isColourBurst && !intraField) {
-        // Examine the other field:
+        // Examine this field:
 
         // Look up the field for a replacement
         findPotentialReplacementLine(thisFieldDropouts, dropOutIndex,
-                                     otherFieldDropouts, false, otherFieldOffset, -stepAmount,
+                                     thisFieldDropouts, true, 0, -stepAmount,
                                      firstActiveFieldLine, lastActiveFieldLine,
-                                     candidates);
+                                     candidates, currentSource, sourceFrameQuality);
 
         // Look down the field for a replacement
         findPotentialReplacementLine(thisFieldDropouts, dropOutIndex,
-                                     otherFieldDropouts, false, otherFieldOffset + stepAmount, stepAmount,
+                                     thisFieldDropouts, true, stepAmount, stepAmount,
                                      firstActiveFieldLine, lastActiveFieldLine,
-                                     candidates);
+                                     candidates, currentSource, sourceFrameQuality);
+
+        // Only check the other field for visible line replacements
+        if (!isColourBurst && !intraField) {
+            // Examine the other field:
+
+            // Look up the field for a replacement
+            findPotentialReplacementLine(thisFieldDropouts, dropOutIndex,
+                                         otherFieldDropouts, false, otherFieldOffset, -stepAmount,
+                                         firstActiveFieldLine, lastActiveFieldLine,
+                                         candidates, currentSource, sourceFrameQuality);
+
+            // Look down the field for a replacement
+            findPotentialReplacementLine(thisFieldDropouts, dropOutIndex,
+                                         otherFieldDropouts, false, otherFieldOffset + stepAmount, stepAmount,
+                                         firstActiveFieldLine, lastActiveFieldLine,
+                                         candidates, currentSource, sourceFrameQuality);
+        }
     }
 
     qDebug() << (isColourBurst ? "Colourburst" : "Visible video") << "dropout on line"
-             << thisFieldDropouts[dropOutIndex].fieldLine << "of" << (thisFieldIsFirst ? "first" : "second") << "field";
+             << thisFieldDropouts[0][dropOutIndex].fieldLine << "of" << (thisFieldIsFirst ? "first" : "second") << "field";
 
     // If no candidate is found, return no replacement
     Replacement replacement;
+    replacement.distance = 0;
 
     if (!candidates.empty()) {
         // Find the candidate with the lowest spatial distance from the dropout
         qint32 lowestDistance = 1000000;
```

Comment:
You're storing the lowest distance as replacement.distance now, so lowestDistance can be removed.

Reply:
Done